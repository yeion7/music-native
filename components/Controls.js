/* @flow weak */

import React from "react";
import { View, Text, StyleSheet } from "react-native";
import styled from "styled-components/native";
import { Button, Icon } from "native-base";

const Controls = ({ onBack, onPlayPause, onNext, isPlaying, index }) => (
  <Container>
    <Button transparent dark onPress={onBack} disabled={index ? false : true}>
      <Icon name="md-skip-backward" />
    </Button>
    {isPlaying
      ? <Button transparent full dark onPress={onPlayPause}>
          <Icon name="md-pause" />
        </Button>
      : <Button transparent full dark onPress={onPlayPause}>
          <Icon name="md-play" />
        </Button>}
    <Button transparent dark onPress={onNext} disabled={index ? false : true}>
      <Icon name="md-skip-forward" />
    </Button>
  </Container>
);

export default Controls;

const Container = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 5;
`;
