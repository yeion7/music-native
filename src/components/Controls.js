/* @flow weak */

import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { Button, Icon } from "native-base";

import styled from "styled-components/native";

const Controls = ({ onBack, onPlayPause, onNext, isPlaying, index }) => (
  <Container>
    {index &&
      <Button transparent dark onPress={onBack}>
        <Icon name="md-skip-backward" />
      </Button>}
    {isPlaying
      ? <Button transparent full dark onPress={onPlayPause}>
          <Icon name="md-pause" />
        </Button>
      : <Button transparent full dark onPress={onPlayPause}>
          <Icon name="md-play" />
        </Button>}
    {index &&
      <Button transparent dark onPress={onNext}>
        <Icon name="md-skip-forward" />
      </Button>}
  </Container>
);

export default Controls;

const Container = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding-vertical: 10;
`;
