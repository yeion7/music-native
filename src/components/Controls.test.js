import React from "react";
import "react-native";

import Controls from "./Controls";

import renderer from "react-test-renderer";

describe("renders test ", () => {
  it("render Controls component playing false and index", () => {
    const tree = renderer
      .create(
        <Controls
          onBack={jest.fn}
          onPlayPause={jest.fn}
          onNext={jest.fn}
          isPlaying="false"
          index={1}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render Controls component playing true and index", () => {
    const tree = renderer
      .create(
        <Controls
          onBack={jest.fn}
          onPlayPause={jest.fn}
          onNext={jest.fn}
          isPlaying="true"
          index={1}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render Controls component playing false and without index", () => {
    const tree = renderer
      .create(
        <Controls
          onBack={jest.fn}
          onPlayPause={jest.fn}
          onNext={jest.fn}
          isPlaying="false"
          index={null}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render Controls component playing true and without index", () => {
    const tree = renderer
      .create(
        <Controls
          onBack={jest.fn}
          onPlayPause={jest.fn}
          onNext={jest.fn}
          isPlaying="true"
          index={null}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
