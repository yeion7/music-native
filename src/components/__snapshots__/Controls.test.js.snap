// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render Controls component playing false and index 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`renders test  render Controls component playing false and without index 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`renders test  render Controls component playing true and index 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`renders test  render Controls component playing true and without index 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": null,
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "elevation": 0,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 6,
        "shadowColor": null,
        "shadowOffset": null,
        "shadowOpacity": null,
        "shadowRadius": null,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
